src/modules/event/services/eventUser/CreateEventUserService.ts

```diff
@@ -0,0 +1,57 @@
+import { inject, injectable } from 'tsyringe';
+import AppError from '@shared/infra/http/errors/AppError';
+import EventUser from '@modules/event/infra/typeorm/entities/EventUser';
+import IDatefnsProvider from '@shared/providers/DatefnsProvider/models/IDatefnsProvider';
+import IEventUserRepository from '../../repositories/IEventUserRepository';
+
+interface IRequest {
+  name: string;
+  date: Date;
+  start_time: Date;
+  end_time: Date;
+  description: string;
+  event_type_id: string;
+  user_id: string;
+}
+
+@injectable()
+export default class CreateEventUserService {
+  constructor(
+    @inject('EventUserRepository')
+    private eventUserRepository: IEventUserRepository,
+
+    @inject('DatefnsProvider')
+    private dateFnsProvider: IDatefnsProvider,
+  ) {}
+
+  public async execute({
+    name,
+    date,
+    start_time,
+    end_time,
+    description,
+    event_type_id,
+    user_id,
+  }: IRequest): Promise<EventUser> {
+    const stringDate = String(date);
+    const thisDateisAfter = await this.dateFnsProvider.thisDateIsAfter(
+      stringDate,
+    );
```

Comment:
```ts
    const thisDateisAfter = await this.dateFnsProvider.thisDateIsAfter(
      String(date)
    );
   ```
   
   do it this way, as there is no need to create a constant in memory just to store stringDate